libraries/support/azure/graph.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'singleton'
+
+module Azure
+  class Graph
+    include Singleton
+
+    def with_client(graph_client, override: false)
+      set_reader(:rest_client, graph_client, override)
+    end
+
+    def for_tenant(tenant_id, override: false)
+      set_reader(:tenant_id, tenant_id, override)
+    end
+
+    def user(id)
+      get(
+        "/#{tenant_id}/users/#{id}",
+        params: { 'api-version' => '1.6' },
+      )
+    end
+
+    def users
+      get(
+        "/#{tenant_id}/users",
+        params: { 'api-version' => '1.6' },
+      )
+    end
+
+    private
+
+    def get(*args)
```

Comment:
This method still feels like it's doing a lot of work. Can we split out the pagination logic to it's own class?